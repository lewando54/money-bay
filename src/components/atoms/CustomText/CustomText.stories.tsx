import { Meta,StoryObj } from '@storybook/react-native'

import CustomText from './CustomText'

const meta = {
    component: CustomText,
    argTypes: {
        // To write
    }
} satisfies Meta<typeof CustomText>

export default meta
type Story = StoryObj<typeof CustomText>

export const Default: Story = {
    args: {
        // To write
    }
}
