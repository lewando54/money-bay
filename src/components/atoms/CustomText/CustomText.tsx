import React from 'react'
import { StyleProp, Text, TextProps, TextStyle } from 'react-native'

import globalThemeStyle from '@styling/GlobalTheme.style'

interface ICustomTextProps extends TextProps {
  children?: React.ReactNode;
  style?: StyleProp<TextStyle>
}

export default function CustomText({ children, style, ...props }: ICustomTextProps) {
    return (
        <Text style={[globalThemeStyle.text_default, style]} {...props}>
            {children}
        </Text>
    )
}
