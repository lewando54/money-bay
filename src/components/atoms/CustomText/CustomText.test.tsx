import React from 'react'
import renderer from 'react-test-renderer'

import CustomText from './CustomText'

import { globalColors } from '@styling/GlobalTheme.style'

describe('CustomText', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<CustomText style={{color: globalColors.BODY_TEXT_COLOR}}>Testing</CustomText>)
        expect(tree).toMatchSnapshot()
    })
})
