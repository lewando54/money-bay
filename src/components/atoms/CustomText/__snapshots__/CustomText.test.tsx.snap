// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText should render correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#1B1D4D",
        "fontFamily": "MulishRegular",
        "fontSize": 16,
      },
      {
        "color": "#4C4C60",
      },
    ]
  }
>
  Testing
</Text>
`;
